tests/tdastro/astro_utils/test_passband_groups.py

```diff
@@ -152,15 +136,59 @@ def test_passband_group_wrapped_from_physical_source(tmp_path):
 
     test_times = np.array([1.0, 1.5, 2.0, 2.5, 3.0, 3.5])
 
-    # Try with a passband group (see Passband tests for single-band tests)
-    test_passband_group = create_passband_group(tmp_path, delta_wave=20, trim_quantile=None)
+    # Test wrapper with a PassbandGroup as input (see Passband tests for single-band tests)
+    # Using LSST passband group:
+    lsst_passband_group = create_lsst_passband_group(passbands_dir, delta_wave=20, trim_quantile=None)
+    result_from_source_model = model.get_band_fluxes(lsst_passband_group, test_times, state)
+    evaluated_fluxes = model.evaluate(test_times, lsst_passband_group.waves, state)
+    result_from_passband_group = lsst_passband_group.fluxes_to_bandfluxes(evaluated_fluxes)
 
-    result_from_source_model = model.get_band_fluxes(test_passband_group, test_times, state)
+    # Check the two dicts are the same
+    assert result_from_source_model.keys() == result_from_passband_group.keys()
+    for key in result_from_source_model:
+        np.testing.assert_allclose(result_from_source_model[key], result_from_passband_group[key])
 
-    evaluated_fluxes = model.evaluate(test_times, test_passband_group.waves, state)
-    result_from_passband_group = test_passband_group.fluxes_to_bandfluxes(evaluated_fluxes)
+    # Using toy passband group:
+    toy_passband_group = create_toy_passband_group(tmp_path, delta_wave=20, trim_quantile=None)
+    result_from_source_model = model.get_band_fluxes(toy_passband_group, test_times, state)
+    evaluated_fluxes = model.evaluate(test_times, toy_passband_group.waves, state)
+    result_from_passband_group = toy_passband_group.fluxes_to_bandfluxes(evaluated_fluxes)
 
     # Check the two dicts are the same
     assert result_from_source_model.keys() == result_from_passband_group.keys()
     for key in result_from_source_model:
         np.testing.assert_allclose(result_from_source_model[key], result_from_passband_group[key])
+
+
+def test_passband_group_calculate_in_band_wave_indices(passbands_dir, tmp_path):
```

Comment:
I'm actually a bit confused about the original function and different return formats for different cases. If a passband waves are represented by a range of `catalog.waves` it returns `(lower_index, upper_index + 1)` tuple, while for the opposite case it is an integer array of indexes. Than here it is used to index an array:
https://github.com/lincc-frameworks/tdastro/blob/7450fccec5d30c5737d2d21772ac8d7421315c12/src/tdastro/astro_utils/passbands.py#L187-L190

I'd suggest to return `slice(lower_index, upper_index + 1)` instead, than both types of the return indexes may be just used directly.

Should I make a separate issue for that?

Reply:
Added